### PR TITLE
[#1149] fix: GC logs in JDK 11 do not include date and time stamps.

### PR DESCRIPTION
**Pull request by qijiale76**
### What changes were proposed in this pull request?

Fix GC logs in JDK 11 do not include date and time stamps

### Why are the changes needed?

Fix: #1149 

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manually tested

### CODE REVIEW

**Comment by qijiale76** [comment]
According to this documentation https://openjdk.org/jeps/158, the correct format should be 
```
-Xlog:gc[:[<output>][:[<decorators>][:<output-options>]]]
```
The previous option missed a `:` symbol, thus `tags,time,uptime,level` is not working

**Comment by jerqi** [comment]
@zhengchenyu Could you help review this pr?

**Comment by qijiale76** [comment]
I will push forward with this PR again this week.

**Comment by jerqi** [comment]
> Do we support higher versions of JDK here, like JDK 21? @qijiale76

We don't support high version now.

**Comment by rickyma** [comment]
> We don't support high version now.

I have run Uniffle on JDK 21 and didn't find any issues. I think Uniffle is compatible with JDK 21 already.

**Comment by jerqi** [comment]
> > We don't support high version now.
> 
> I have run Uniffle on JDK 21 and didn't find any issues. I think Uniffle is compatible with JDK 21 already.

We should add a JDK 21 pipeline first.

**Comment by EnricoMi** [comment]
@qijiale76 can this go ahead?

**Comment by EnricoMi** [comment]
@qijiale76 @zhengchenyu @jerqi is this ready for review?

**Comment by zhengchenyu** [comment]
I don't think we need to support all versions of the JDK, the user should configure it according to their own environment.
But the docker image is using JDK11, the pom is using JDK8. Then I don't know which version to support.

**Comment by jerqi** [comment]
> I don't think we need to support all versions of the JDK, the user should configure it according to their own environment. But the docker image is using JDK11, the pom is using JDK8. Then I don't know which version to support.

You can see the pipelines to find which version to support. Spark supports JDK8, JDK11 and JDK17.

**Comment by zhengchenyu** [comment]
> > I don't think we need to support all versions of the JDK, the user should configure it according to their own environment. But the docker image is using JDK11, the pom is using JDK8. Then I don't know which version to support.
> 
> You can see the pipelines https://github.com/apache/incubator-uniffle/blob/master/.github/workflows/build.yml to find which version to support. Spark supports JDK8, JDK11 and JDK17.

OK, we should support JDK17!

**Comment by qijiale76** [comment]
I just fixed the issue causing the test failure. I'll test the Java 17 configuration tomorrow.

**Comment by jerqi** [comment]
@qijiale76 Could you update the description?
@zhengchenyu @xianjingfeng Could you help me review this pull request?

**Comment by zhengchenyu** [comment]
I'm not very familiar with later Java GC configurations. If only for JDK 8 and 11, LGTM+1!

**Review by jerqi — APPROVED** [review]

**Review by zhengchenyu — APPROVED** [review]

File: bin/start-shuffle-server.sh [hunk 1]
```diff
@@ -103,22 +103,30 @@ JVM_ARGS=" -server \
...
           -XX:+PrintPromotionFailure \
           -XX:+PrintGCApplicationStoppedTime \
-          -XX:+PrintCommandLineFlags \
           -XX:+PrintGCCause \
           -XX:+PrintGCDetails \
           -Xloggc:${RSS_LOG_DIR}/gc-%t.log"
 
-JAVA11_EXTRA_ARGS=" -XX:+IgnoreUnrecognizedVMOptions \
-          -Xlog:gc:tags,time,uptime,level"
+GC_LOG_ARGS_NEW=" -XX:+IgnoreUnrecognizedVMOptions \
```
Comment (LuciferYang) [inline bin/start-shuffle-server.sh:121]: Can we remove `-XX:+IgnoreUnrecognizedVMOptions` after this pr?

File: bin/start-coordinator.sh [hunk 2]
```diff
@@ -68,20 +68,28 @@ JVM_ARGS=" -server \
...
           -XX:+PrintPromotionFailure \
           -XX:+PrintGCApplicationStoppedTime \
-          -XX:+PrintCommandLineFlags \
           -XX:+PrintGCCause \
           -XX:+PrintGCDetails \
           -Xloggc:${RSS_LOG_DIR}/gc-%t.log"
 
-JAVA11_EXTRA_ARGS=" -XX:+IgnoreUnrecognizedVMOptions \
-          -Xlog:gc:tags,time,uptime,level"
+GC_LOG_ARGS_NEW=" -XX:+IgnoreUnrecognizedVMOptions \
```
Comment (LuciferYang) [inline bin/start-coordinator.sh:84]: Can we remove -XX:+IgnoreUnrecognizedVMOptions after this pr?

File: bin/uniffle [hunk 3]
```diff
@@ -60,16 +60,27 @@ function uniffle_cmd_case
...
+      GC_LOG_ARGS_LEGACY=" -XX:+PrintGC \
           -XX:+PrintAdaptiveSizePolicy \
           -XX:+PrintGCDateStamps \
           -XX:+PrintGCTimeStamps \
           -XX:+PrintGCDetails \
           -Xloggc:${RSS_LOG_DIR}/gc-%t.log"
 
-      JAVA11_EXTRA_ARGS=" -XX:+IgnoreUnrecognizedVMOptions \
-          -Xlog:gc:tags,time,uptime,level"
+      GC_LOG_ARGS_NEW=" -XX:+IgnoreUnrecognizedVMOptions \
```
Comment (LuciferYang) [inline bin/uniffle:72]: ditto

**Pull request merged**